engines/private/funcs.cpp

```diff
@@ -0,0 +1,784 @@
+    }
+}
+
+void fViewScreen(ArgArray args) {
+    // assert types
+    debugC(1, kPrivateDebugScript, "WARNING: ViewScreen not implemented!");
+}
+
+void fTransition(ArgArray args) {
+    // assert types
+    debugC(1, kPrivateDebugScript, "Transition(%s, %s)", args[0].u.str, args[1].u.str);
+    g_private->_nextMovie = new Common::String(args[0].u.str);
+    g_private->_nextSetting = new Common::String(args[1].u.str);
+}
+
+void fResume(ArgArray args) {
+    // assert types
+    debugC(1, kPrivateDebugScript, "Resume(%d)", args[0].u.val); // this value is always 1
+    g_private->_nextSetting = g_private->_pausedSetting;
+    g_private->_pausedSetting = NULL;
+    g_private->_mode = 1;
+    g_private->setOrigin(kOriginOne);
+}
+
+void fMovie(ArgArray args) {
+    // assert types
+    debugC(1, kPrivateDebugScript, "Movie(%s, %s)", args[0].u.str, args[1].u.str);
+    Common::String *movie = new Common::String(args[0].u.str);
+    Common::String *nextSetting = new Common::String(args[1].u.str);
+    bool isEmptyString = strcmp(args[0].u.str, "\"\"") == 0;
+
+    if (!g_private->_playedMovies.contains(*movie) && !isEmptyString) {
+        g_private->_nextMovie = movie;
+        g_private->_playedMovies.setVal(*movie, true);
+        g_private->_nextSetting = nextSetting;
+
+    } else if (isEmptyString) {
+        g_private->_repeatedMovieExit = nextSetting;
+        debugC(1, kPrivateDebugScript, "repeated movie exit is %s", nextSetting->c_str());
+    } else {
+        debugC(1, kPrivateDebugScript, "movie %s already played", movie->c_str());
+        g_private->_nextSetting = g_private->_repeatedMovieExit;
+    }
+}
+
+void fCRect(ArgArray args) {
+    // assert types
+    debugC(1, kPrivateDebugScript, "CRect(%d, %d, %d, %d)", args[0].u.val, args[1].u.val, args[2].u.val, args[3].u.val);
+    int x1, y1, x2, y2;
+    x1 = args[0].u.val;
+    y1 = args[1].u.val;
+    x2 = args[2].u.val;
+    y2 = args[3].u.val;
+
+    Datum *d = new Datum();
+    Common::Rect *rect = new Common::Rect(x1, y1, x2, y2);
+    d->type = RECT;
+    d->u.rect = rect;
+    push(*d);
+}
+
+void fBitmap(ArgArray args) {
+    assert(args.size() == 1 || args.size() == 3);
+
+    int x = 0;
+    int y = 0;
+
+    char *f = args[0].u.str;
+    if (args.size() == 3) {
+        x = args[1].u.val;
+        y = args[2].u.val;
+    }
+
+    debugC(1, kPrivateDebugScript, "Bitmap(%s, %d, %d)", f, x, y);
+    Common::String *s = new Common::String(args[0].u.str);
+    g_private->loadImage(*s, x, y);
+}
+
+void _fMask(ArgArray args, bool drawn) {
+    assert(args.size() == 3 || args.size() == 5);
+
+    int x = 0;
+    int y = 0;
+    char *f = args[0].u.str;
+    char *e = args[1].u.str;
+    Common::String *c = args[2].u.sym->name;
+
+    if (args.size() == 5) {
+        x = args[3].u.val;
+        y = args[4].u.val;
+    }
+
+    debugC(1, kPrivateDebugScript, "Mask(%s, %s, %s, %d, %d)", f, e, c->c_str(), x, y);
+    const Common::String *s = new Common::String(f);
```

Comment:
Same construct. You are creating here an object but never free it. Then, you are not changing it in the called methods, `loadMask()` in this case, thus, there is no need to do this creation at all: just pass it by reference.

Reply:
Fixed